docs/diagrams/apiml-oidc-auth-seq.puml

```diff
@@ -4,32 +4,37 @@
 autonumber
 actor User
 actor "User Agent" as Agent
-actor "Client App" as Client
 actor OIDC as OIDC
 actor "API ML GW" as GW
 
 User -> Agent: Do stuff
-Agent -> Client: Open Client App
-Client -> OIDC: Initiate OIDC flow [client_id, client_secret]
-loop [MFA]
+Agent -> GW: /gateway/oauth2/authorization/<provider-id>
+GW -> OIDC: Initiate OIDC flow [client_id, client_secret]
+group MFA
 OIDC -> Agent: Request user credentials
 User <-> Agent: Provide credentials
 Agent -> OIDC: Validate credentials
 end
-OIDC --> Client: Return Auth Code / JWTs[access,refresh,identity]
-Client -> GW: Request resources [JWT AT]
-group validate
-GW -> GW: Validate AT
-GW -> OIDC: Validate AT
-end
-alt Validation success
-GW -> GW: Cache AT validity
+OIDC --> GW: Return Auth Code / JWTs[access,refresh,identity]
+GW -> Agent: set-cookie access token
+Agent -> GW: Request resources [access token]
+GW -> GW: Validate access token
+GW -> GW: Cache access token validity
+group map user
 GW -> SAF: Map distributed ID to mainframe ID
+end
+alt mapping success
 GW -> GW: Create Zowe JWT
-GW -> Service: call API service
+GW -> Service: call API service with Zowe JWT
+Service -> Service: Validate JWT
 Service --> GW: return Response
 end
-GW --> Client: Response
-Client --> Agent: Response
+alt mapping failure
+GW -> Service: call API service with access token
+Service -> Service: Validate access token
+Service --> GW: return Response
+end
```

Comment:
This is a correct syntax for plantuml alternative group:

@startuml
Alice -> Bob: Authentication Request
alt Successful case
    Bob -> Alice: Authentication Accepted
else Failure
    Bob -> Alice: Authentication Rejected
end
@enduml